lib/block-supports/layout.php

```diff
@@ -136,28 +148,31 @@ function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support
 			 * by custom css.
 			 */
 			if ( ! empty( $layout['justifyContent'] ) && array_key_exists( $layout['justifyContent'], $justify_content_options ) ) {
-				$style .= "$selector {";
-				$style .= "justify-content: {$justify_content_options[ $layout['justifyContent'] ]};";
-				$style .= '}';
+				$store->add_rule( $selector )->set_declarations( array( 'justify-content' => $justify_content_options[ $layout['justifyContent'] ] ) );
 			}
 
 			if ( ! empty( $layout['verticalAlignment'] ) && array_key_exists( $layout['verticalAlignment'], $vertical_alignment_options ) ) {
-				$style .= "$selector {";
-				$style .= "align-items: {$vertical_alignment_options[ $layout['verticalAlignment'] ]};";
-				$style .= '}';
+				$store->add_rule( $selector )->set_declarations( array( 'align-items' => $vertical_alignment_options[ $layout['verticalAlignment'] ] ) );
 			}
 		} else {
-			$style .= "$selector {";
-			$style .= 'flex-direction: column;';
+			$store->add_rule( $selector )->set_declarations(
+				array(
+					'flex-direction' => 'column',
+					'align-items'    => 'flex-start',
+				)
+			);
 			if ( ! empty( $layout['justifyContent'] ) && array_key_exists( $layout['justifyContent'], $justify_content_options ) ) {
-				$style .= "align-items: {$justify_content_options[ $layout['justifyContent'] ]};";
-			} else {
-				$style .= 'align-items: flex-start;';
+				$store->add_rule( $selector )->set_declarations( array( 'align-items' => $justify_content_options[ $layout['justifyContent'] ] ) );
 			}
-			$style .= '}';
 		}
 	}
 
+	$style = '';
+	$rules = $store->get_all_rules();
+	foreach ( $rules as $rule ) {
+		$style .= $rule->get_css();
+	}
+
```

Comment:
This part will be replaced by the `Processor`/`Renderer` implementation when that gets merged

Reply:
This might not be ideal since we'd be enqueuing block supports styles from many places. Wouldn't this pull out all of them, not just layout-related?

This is why I think having an abstracted method, that uses the classes in the background, and that centralizes the work, could be a better option. 

Example: https://github.com/WordPress/gutenberg/pull/42452/files#diff-69616e5e4ff1a06362db0b7165c330b00614d4ea0d3a28c80003c0860ab8b93fR683

Reply:
Hmmm you're right. We'll need to make sure that all these styles only get printed once 👍

Reply:
What if we make the store names a bit more specific for the time being until we have a more abstract & generalized method to handle these? So instead of having a `block-supports` store, we could name it `block-supports/layout`. This would make it more specific and each store can be handled differently until we have a more generic strategy 🤔

Reply:
True, it's a bit tricky. I think we'll have to experiment a bit before we get it right. 🤔